src/Microsoft.AspNetCore.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -325,13 +332,17 @@ private void WriteAllPending()
 
             if (Monitor.TryEnter(_contextLock))
             {
-                _writePending = false;
+                _postingWrite = false;
 
                 if (_nextWriteContext != null)
                 {
                     writingContext = _nextWriteContext;
                     _nextWriteContext = null;
                 }
+                else
+                {
+                    _ongoingWrites--;
```

Comment:
I had those initially, but I noticed that I was calling it in between Monitor.TryEnter and Monitor.Exit and stuff which was a little scary. They shouldn't fail, and they didn't in the 10 or so times I ran through the unit and functional tests. Once I added SocketOutput tests, I felt it was unnecessary.